test/unit/streaming.ts

```diff
@@ -306,4 +306,150 @@ describe('streaming', () => {
       done();
     });
   });
+
+  it('emit response when stream received metadata event', done => {
+    const responseMetadata = {metadata: true};
+    const expectedStatus = {code: 0, metadata: responseMetadata};
+    const expectedResponse = {
+      code: 200,
+      message: 'OK',
+      details: '',
+      metadata: responseMetadata,
+    };
+    const spy = sinon.spy((...args: Array<{}>) => {
+      assert.strictEqual(args.length, 3);
+      const s = new PassThrough({
+        objectMode: true,
+      });
+      s.push(null);
+      setImmediate(() => {
+        s.emit('metadata', responseMetadata);
```

Comment:
I'm curious if metadata happened to be a huge buffer, or if this was happening on a slow connectoin, is there a chance `status` would emit before `metadata` finished reading?

Reply:
According to @murgatroid99, the `status` event is always sent after the last message.
I assume the `metadata` to be emitted early before `status`. However, if the `status` sent earlier, then the `response` emit from `status` event.

Reply:
If there is a `metadata` event, it is always emitted first. If there are any `data` events, they are emitted after `metadata`. The `status` event is always emitted, and it is always after the others.